torch/csrc/distributed/rpc/agent_utils.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+
+#include <c10d/PrefixStore.hpp>
+#include <torch/csrc/distributed/rpc/utils.h>
+
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+// All RPC peers should call into this function at the same time. Each peer
+// provides its own id and name, and this function uses the given Store to
+// gather global name-to-id mapping on all peers.
+std::unordered_map<std::string, worker_id_t> collectNames(
+    ::c10d::PrefixStore store,
+    const worker_id_t selfId,
+    const std::string& selfName,
+    const int worldSize) {
+  std::vector<uint8_t> selfNameVector(
+      (uint8_t*)selfName.c_str(),
+      (uint8_t*)selfName.c_str() + selfName.length());
+  store.set(c10::to_string(selfId), selfNameVector);
+
+  std::unordered_map<std::string, worker_id_t> nameToId;
+  nameToId.reserve(worldSize);
+  nameToId.emplace(selfName, selfId);
+  for (worker_id_t workerId = 0; workerId < worldSize; ++workerId) {
+    if (workerId == selfId) {
+      continue;
+    }
+    std::vector<uint8_t> workerNameVector = store.get(c10::to_string(workerId));
+    std::string workerName(
+        (char*)workerNameVector.data(), workerNameVector.size());
+
+    TORCH_CHECK(
+        nameToId.find(workerName) == nameToId.end(),
+        "RPC worker name ",
+        workerName,
+        " is not unique.");
```

Comment:
Nit: to help with debugging, it might be useful to log the ids for the duplicate names here: `nameToId.find(workerName)->second` and `workerId`.

Reply:
good point, will add.

Reply:
will add this to the top PR to avoid modifying this twice during rebase.